examples/gno.land/r/x/manfred_upgrade_patterns/upgrade_g/logic/v1/counter.gno

```diff
@@ -0,0 +1,29 @@
+package v1
+
+import (
+	"std"
+
+	"gno.land/r/x/manfred_upgrade_patterns/upgrade_g/admin"
+)
+
+var store admin.Store = nil
```

Comment:
Why does `store` need to be the global, and not an instance of `logic`?

Reply:
The main question is why doesn't this `store` instance live in the `logic` instance?

Reply:
The `logic` is simply a set of functions to be registered to the admin. It is a workaround with `interface`s as they are not able to cross the realm boundary. Store, in this case, needs to be accessed by any functions in the realm, entrypoint or not. By having store as a top level variable it could have the similar coding style with having state variables.